website/docs/docs/deploy/job-scheduler.md

```diff
@@ -0,0 +1,99 @@
+| Run slot | Run slots control the number of jobs that can run concurrently. Each account has a fixed number of run slots, depending on the plan tier, that are shared across projects in the account. Each running job occupies a run slot for the duration of the run, so purchasing more run slots enables more jobs to execute in parallel. |
+| Threads | When dbt builds a project's DAG, it tries to parallelize the execution by using threads. The [thread](/docs/core/connection-profiles#understanding-threads) count is the maximum number of paths through the DAG that dbt can work on simultaneously. The default value is 4 threads. |
+| Wait time | Amount of time that dbt Cloud waits before running a job, either because there are no available slots or because a previous run of the same job is still in progress. |
+
+
+## Scheduler queue
+
+The dbt Cloud scheduler queues a deployment job to be processed when it's triggered to run, either by schedule, by API, by PR trigger, or manually. 
+
+Before the job starts executing, the scheduler checks two conditions to determine if the run can start executing:
+
+- **The account must have an available run slot** &mdash; If all run slots are occupied, the queued run will wait. The wait time is displayed in dbt Cloud. If there are long wait times, [upgrading to enterprise](https://www.getdbt.com/contact/) can provide more run slots and allow for higher job concurrency.
+
+- **There must not be a run of the same job already in-flight** &mdash; The Scheduler executes distinct runs of the same dbt Cloud job serially to avoid model build collisions. If there's a job already running, the queued job will wait, and the wait time will be displayed in dbt Cloud.
+
+If there is an available run slot and there's not an actively running instance of the job, the Scheduler will prepare the job to run in the user's cloud data platform. This prep involves readying a Kubernetes pod with the right version of dbt installed, setting environment variables, loading data platform credentials, and git provider authorization, amongst other environment-setting tasks. The time it takes to prepare the job is displayed as **prep time.**
+
+Collectively **wait time** and **prep time** is the time a run spends in queue (or **Time in queue**).
+
+<Lightbox src="/img/docs/dbt-cloud/deployment/deploy-scheduler.jpg" width="85%" title="An overview of a dbt Cloud job run"/>
+
+## Job memory
+
+In dbt Cloud, each run of a job executes in a pod with a default megabyte (MB) memory limit. If a job uses up all of the memory, the pod is terminated with a "memory limit error" message. There isn't a memory limit set at the account-level.
+
+Jobs consume a lot of memory in the following situations:
+- A high thread count was specified
+- Custom dbt macros attempt to load data into memory instead of pushing compute down to the cloud data platform
+- Generating dbt project documentation when a source's schema has a large number of tables
```

Comment:
feeback from @will-sargent-dbtlabs 

```suggestion
- Having a job that generates dbt project documentation for a large and complex dbt project. To prevent problems with the job running out of memory, we recommend generating documentation in a separate job that is set aside for that task and removing that task from all other jobs. This is especially important for large and complex projects.
```